src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: [], key: string): GroupsMap<Object> {
+  const group: GroupsMap<Object> = {};
+
+  items.forEach((item) => {
+    if (item[key]) {
+      if (!group[item[key]]) {
+        group[item[key]] = [];
+      }
+    }
+    group[item[key]].push(item);
```

Comment:
Pushing the item into the group[item[key]] array is done outside of the if block that checks if 'item[key]' exists. This could potentially lead to errors if 'item[key]' doesn't exist. You should move this line inside the if block.